File: front/src/domain-types/models/usuario.ts
export type Usuario = {
  id: number;
  nome: string;
  email: string;
};

export type LoginUsuarioFormInput = {
  email: string;
  senha: string;
};

export type CadastroUsuarioFormInput = {
  nome: string;
  email: string;
  senha: string;
  confirmarSenha: string;
};

File: backend/GLBViewerAPI/Domain/Models/Usuario.cs
using FluentValidation.Results;
using GLBViewerAPI.Domain.Validators;
using GLBViewerAPI.Domain.Contracts;
using GLBViewerAPI.Domain.Models;

namespace GLBViewerAPI.Models
{
    public class Usuario : Entity, IAggregateRoot
    {
        public int Id { get; set; }
        public string Nome { get; set; } = null!;
        public string Email { get; set; } = null!;
        public string Senha { get; set; } = null!;

        public override bool Validar(out ValidationResult validationResult)
        {
            validationResult = new UsuarioValidator().Validate(this);
            return validationResult.IsValid;
        }
    }
}
